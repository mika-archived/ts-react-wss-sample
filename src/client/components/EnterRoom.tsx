import React, { useState } from "react";
import PropTypes from "prop-types";

import Button from "./controls/Button";
import TextBox from "./controls/TextBox";

type Props = {
  onSubmit: (id: string) => void;
};

const EnterRoom: React.FC<Props> = props => {
  const [value, setValue] = useState<string>("");

  return (
    <>
      <p>ルーム ID を入力してください。</p>
      <div>
        <TextBox type="text" onChange={event => setValue(event.target.value)} />
        <Button primary type="submit" disabled={!value} onClick={() => props.onSubmit(value)}>
          入室する
        </Button>
      </div>
    </>
  );
};

EnterRoom.propTypes = {
  onSubmit: PropTypes.func.isRequired
};

export default EnterRoom;
