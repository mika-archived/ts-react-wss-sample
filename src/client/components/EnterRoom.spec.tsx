import "@testing-library/jest-dom/extend-expect";

import React from "react";
import { cleanup, fireEvent, render } from "@testing-library/react";
import EnterRoom from "./EnterRoom";

afterEach(cleanup);

test("does not have changes between snapshots", () => {
  const { asFragment } = render(<EnterRoom onSubmit={() => {}} />);

  expect(asFragment()).toMatchSnapshot();
});

test("fire `onSubmit` event when enter the room id and click the button", () => {
  const callback = jest.fn();
  const { getByRole } = render(<EnterRoom onSubmit={callback} />);

  fireEvent.change(getByRole("textbox"), { target: { value: "test" } });
  fireEvent.click(getByRole("button"));

  expect(callback).toHaveBeenCalledTimes(1);
});

test("if room id is empty, can not click (fire) the submit button", () => {
  const callback = jest.fn();
  const { getByRole } = render(<EnterRoom onSubmit={callback} />);

  fireEvent.click(getByRole("button"));

  expect(callback).toHaveBeenCalledTimes(0);
});
