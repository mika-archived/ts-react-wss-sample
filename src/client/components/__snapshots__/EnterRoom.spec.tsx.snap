// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not have changes between snapshots 1`] = `
<DocumentFragment>
  <p>
    ルーム ID を入力してください。
  </p>
  <div>
    <input
      class="sc-bwzfXH eodmcB"
      role="textbox"
      type="text"
    />
    <button
      class="sc-bdVaJa ldHTCk"
      disabled=""
      role="button"
      type="submit"
    >
      入室する
    </button>
  </div>
</DocumentFragment>
`;
